### ISSUE TITLE
How to peer discover without using socket.io

### ISSUE THREAD

**Issue by macdasi** [peers/peerjs-server#98]
Is there a way to let know other peers (A) when a peer (B) is connected / disconnected ? 
With regard to : 
A don't know B peerid .
I don't want to use another websocket like socket.io 
I think ExpressPeerServer has members that control socket to peers , so all I got to do is iterate each one and emit a message ... I can't find a way to do it . 
I don't want the client to make xhr request for peers list or use peer.listAllPeers()

**Comment by ederollora** [comment]
If you don't want to use Websockets or AJAX then how do you want to let clients know about other clients? It does not make any sense. You could always define a path (/otherclients) and answer with the required info to any GET request to that path, following a RESTful paradigm. But you would neet the client to poll the server every n seconds. Alternatively, you can implement the same but accepting POST requests were the client specifies a parameter it knows in the body and the server answers back the id of the other peer based on the parameter you specified. But if you want a way were a client listens and the server sends some info based on an event regarding another user, then I cannot imagine that situation without WebSockets although let us know if you come with a better solution.

**Comment by luongnv89** [comment]
@macdasi You can create a virtual peer which will work like a NETWORK OPERATOR (OP - has a fixed ID). So every peer will first connect with OP, then OP will provide the information of other peers (like list of peer ids), then the new peer can connect with another peer (it also can keep connect with OP if you want).
I have not implemented this idea yet, just a pop idea in my head.

**Comment by macdasi** [comment]
I have a pull request 99, changed the peer-server implementation to had support for signaling other connected peers. As I see it the peer-server already has a socket open to each client so it can send message to each peer to notify on new peer connection with the peerId as payload. 
My change was due to the fact I didn't want to open another socket or create additional XHR requests

**Comment by ederollora** [comment]
Yes I saw it but I got a couple of questions, I may write them in the request since it may make more sense, is that ok?
